packages/template-retail-react-app/app/pages/account/index.jsx

```diff
@@ -107,14 +107,19 @@ const Account = () => {
         navigate('/login')
     }
 
+    // Do not render anything in the account page on the server!
+    if (!onClient) {
```

Comment:
If we no longer use the `document` object on the server, it should be ok to server side render the account page right? Asking because it seems that we're seeing lots of hydration errors

![Screenshot 2024-02-22 at 5 04 30 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/88680517/8b1d8716-860b-4474-8e4d-22e7dc4ec912)

Reply:
I can't remember if we were always rendering the account page on the server or if it was added by mistake, but by nature all the data that it displays is private customer data and shouldn't be rendered on the client. 

So this is one of the isomorphic development problems that you run into here and there. Because you are always a guest on the server you would be redirected to the login page which would be rendered and sent to the client. Upon loading the app on the client, it would see that you are registered and send you to the account page. So you would get this login page > account page flow. 

If you leave the code you commented on in, then you simply have nothing rendered, then the account page. Those errors that you see are only ever shown in develop mode and not in production. 

But I guess it all depends on what the behaviour was before the introduction of that breaking change.

Do you happen to know what that was?

Reply:
From my understanding, before the breaking change we would render the static account page on the server and then hydrate once it got to the client. I don't see any logic that prevents server side rendering in the previous version: https://github.com/SalesforceCommerceCloud/pwa-kit/blob/0843abc9c9dd05d3d9f3a4a2aab211dae4940aa9/packages/template-retail-react-app/app/pages/account/index.jsx

Seeing the login page then account page if you're a registered user doesn't seem like good UX, nothing rendered then account page is probably better. If these errors are only shown development and not production, then I think we should leave in that logic to prevent rendering on the server